test/jdk/tools/jar/ManifestDirectoryCompression.java

```diff
@@ -48,38 +51,43 @@ public class ManifestDirectoryCompression {
             ToolProvider.findFirst("jar")
                     .orElseThrow(() -> new RuntimeException("jar tool not found"));
 
+    private Path tempDir;
+
+    @BeforeMethod
+    private void setUp() throws Exception {
+        tempDir = Files.createTempDirectory("temp");
+    }
+
     /** Remove dirs & files needed for test. */
-    private static void cleanup(Path dir) {
+    @AfterMethod
+    private void cleanup() {
+        deleteRecursively(tempDir);
+    }
+
+    private static void deleteRecursively(Path path) {
         try {
-            if (Files.isDirectory(dir)) {
-                try (Stream<Path> s = Files.list(dir)) {
-                    s.forEach(p -> cleanup(p));
+            if (Files.isDirectory(path)) {
+                try (Stream<Path> s = Files.list(path)) {
+                    s.forEach(p -> deleteRecursively(p));
                 }
             }
-            Files.delete(dir);
+            Files.delete(path);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
     @Test
-    public void run() throws Throwable {
-        Path topDir = Files.createTempDirectory("delete");
-        try {
-            Path entry = Files.writeString(topDir.resolve("test.txt"), "Some text...");
-            doTest(topDir.resolve("test.jar"), entry);
-        } finally {
-            cleanup(topDir);
-        }
-    }
-
-    private static void doTest(Path jar, Path entry) throws Throwable {
-        String[] jarArgs = new String[] {"cf", jar.toString(), entry.toString()};
+    public void run() throws Exception {
```

Comment:
Please rename `run() `to something like `TestDirectoryCompressionMethod()`.  We are trying to make new tests have more meaningful names.

Reply:
Done